ona/tasks.py

```diff
@@ -0,0 +1,59 @@
+from celery.task import Task
+from celery.utils.log import get_task_logger
+from celery.exceptions import SoftTimeLimitExceeded
+from django.core.exceptions import ObjectDoesNotExist
+from go_http.send import HttpApiSender
+import requests, base64, json
+from requests.exceptions import HTTPError
+from .models import Submission
+
+try:
+    from HTMLParser import HTMLParser
+except ImportError:
+    from html.parser import HTMLParser
+
+logger = get_task_logger(__name__)
+
+class Send_Submission(Task):
+
+    """
+    Task to load and construct submission and send it off
+    """
+    name = "ona.tasks.send_submission"
+
+    class FailedEventRequest(Exception):
+
+        """
+        The attempted task failed because of a non-200 HTTP return code.
+        """
+
+    def run(self, submission_id, **kwargs):
+        try:
+            submission = Submission.objects.get(pk=submission_id)
+            if submission.submitted is False:
+                integration = submission.integration.details
+                data = '{"submission": '+submission.content+', "id": "'+integration["form_id"]+'"}'
+                print data
+                base64string = base64.encodestring(
+                    '%s:%s' % (integration["username"],
+                    integration["password"])).replace('\n', ''
+                )
```

Comment:
This might help: https://ona.io/static/docs/index.html#authentication , although mostly we've been using basic auth over https with all our Ona things so far, there is the option to use OAuth2.